### PR TITLE
Make Resistance reduce Fire, make player applied Fire give damage/kill credit

### LINKED ISSUE #1024: Skills that give fire do not give kill/damage credit

### LINKED ISSUE #1044: Resistance doesnt lessen the duration of fire

### PR DESCRIPTION
**Pull request by StudentAlleg**
Make resistance reduce duration of fire,
Change all instances of set fire ticks to use UtilEntity.setFire, which sets fire and gives damage credit

Fixes #1044
Fixes #1024

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested my changes.

### CODE REVIEW

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/global/Resistance.java [hunk 1]
```diff
@@ -94,6 +96,20 @@ public void onPotionEffectReceived(EntityPotionEffectEvent event) {
         }
     }
 
+    @EventHandler
+    public void onPotionEffectReceived(EntityCombustByEntityEvent event) {
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/global/Resistance.java:100]: Function name

File: core/src/main/java/me/mykindos/betterpvp/core/utilities/UtilEntity.java [hunk 2]
```diff
@@ -180,4 +181,17 @@ public static void setViewRangeBlocks(@NotNull Display display, float blocks) {
...
+
+    /**
+     * Sets fire to the damagee with damager as the source
+     * @param damagee The entity to set on Fire
+     * @param damager The entity that set the fire
+     * @param duration The duration to set the damagee on fire, in ms
+     */
+    public static void setFire(@NotNull Entity damagee, @NotNull Entity damager, long duration) {
+        EntityCombustByEntityEvent entityCombustByEntityEvent = new EntityCombustByEntityEvent(damagee, damager, (float) duration /1000L);
+        UtilServer.callEvent(entityCombustByEntityEvent);
```
Comment (Mykindos) [inline core/src/main/java/me/mykindos/betterpvp/core/utilities/UtilEntity.java:193]: You can do entityCombustByEntityEvent = UtilServer.callEvent(new event...) to save yourself a line here

**Review by Mykindos — CHANGES_REQUESTED** [review]

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/global/Resistance.java (hunk 1 above)
Comment (StudentAlleg) [reply champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/global/Resistance.java]: Changed

File: core/src/main/java/me/mykindos/betterpvp/core/utilities/UtilEntity.java (hunk 2 above)
Comment (StudentAlleg) [reply core/src/main/java/me/mykindos/betterpvp/core/utilities/UtilEntity.java]: done

**Review by Mykindos — APPROVED** [review]

**Pull request merged**